redis/tests/test_async.rs

```diff
@@ -781,3 +782,40 @@ mod mtls_test {
         }
     }
 }
+
+#[test]
+fn test_set_client_name_by_config() {
+    const CLIENT_NAME: &str = "TEST_CLIENT_NAME";
+    use redis::RedisError;
+    let ctx = TestContext::with_client_name(CLIENT_NAME);
+
+    block_on_all(async move {
+        let mut con = ctx.async_connection().await?;
+
+        let client_info: String = redis::cmd("CLIENT")
+            .arg("INFO")
+            .query_async(&mut con)
+            .await
+            .unwrap();
+
+        let client_attrs: Vec<&str> = client_info.split(' ').collect();
+        for attr in client_attrs.iter() {
+            let this_attr: Vec<&str> = attr.split('=').collect();
+            if this_attr[0] == "name" {
+                assert!(
```

Comment:
you can use `assert_eq!`, it logs the different values so there's no need for the extra message.